### PR TITLE
implemented scroll

### PR DESCRIPTION
**Pull request by Neemahj**

### CODE REVIEW

File: src/components/BlogPage.js [hunk 1]
```diff
@@ -1,49 +1,98 @@
...
 const BlogPage = () => {
   const [displayBlogs, setDisplayBlogs] = useState([]);
+  const [page, setPage] = useState(1);
+  const [loading, setLoading] = useState(false);
+  const [noData, setNoData] = useState(false);
+
+  const getBlog = async (page) => {
+    try {
+      let url;
+      if ((page != null) && (page > 1)) {
```
Comment (Asylcreek) [inline src/components/BlogPage.js]: Is `page` ever going to be `null`?

File: src/components/BlogPage.js [hunk 2]
```diff
@@ -1,49 +1,98 @@
...
+  window.onscroll = () => {
+    if (
+      window.innerHeight + document.documentElement.scrollTop ===
+      document.documentElement.offsetHeight
+    ) {
+      if (!noData) {
+        loadBlogs(page);
+      }
+    }
+  }
```
Comment (Asylcreek) [inline src/components/BlogPage.js]: Rewrite this without tying it to the `window` object.

Use event listeners in your `useEffect`

File: src/components/BlogPage.js [hunk 3]
```diff
@@ -1,49 +1,98 @@
...
+  },[]);
+
+  const loadBlogs = (page) => {
+    setLoading(true);
+    setTimeout(() => {
+      getBlog(page)
+          setDisplayBlogs((previousPost)=> [...previousPost, ...displayBlogs])
+          const newPage = page + 1;
+          setPage(newPage);    
+          if(!(displayBlogs)){
```
Comment (Asylcreek) [inline src/components/BlogPage.js]: `displayBlogs` is array by default, it is always **truthy**, you should be checking it's length instead

File: src/components/BlogPage.js [hunk 4]
```diff
@@ -1,49 +1,98 @@
...
-      } catch {
-        console.log("ERROR");
+      loadBlogs(page);
+  },[]);
+
+  const loadBlogs = (page) => {
+    setLoading(true);
+    setTimeout(() => {
+      getBlog(page)
+          setDisplayBlogs((previousPost)=> [...previousPost, ...displayBlogs])
```
Comment (Asylcreek) [inline src/components/BlogPage.js]: If `getBlog` already sets `displayBlogs` why then are you calling `setDisplayBlogs` again?

File: src/components/BlogPage.js [hunk 5]
```diff
@@ -1,49 +1,98 @@
...
-    const getBlog = async () => {
-      try {
-        const { data } = await axios.get(API_URL);
-        setDisplayBlogs(data.getBlogPost.docs);
-      } catch {
-        console.log("ERROR");
+      loadBlogs(page);
+  },[]);
+
+  const loadBlogs = (page) => {
```
Comment (Asylcreek) [inline src/components/BlogPage.js]: Move `loadBlogs` into the `useEffect`

**Review by Asylcreek — CHANGES_REQUESTED** [review]
Keep in mind that you are not preventing fetching blog posts when there are no more blog posts to fetch

File: src/components/BlogPage.js (hunk 5 above)
Comment (Neemahj) [reply src/components/BlogPage.js]: okay sir

File: src/components/BlogPage.js (hunk 1 above)
Comment (Neemahj) [reply src/components/BlogPage.js]: yes sir

**Review by Asylcreek — APPROVED** [review]

**Pull request merged**